ddtrace/tracer.py

```diff
@@ -196,23 +197,31 @@ def global_excepthook(self, tp, value, traceback):
         """The global tracer except hook."""
 
     def get_call_context(self, *args, **kwargs):
-        # type: (...) -> Context
-        """
-        Return the current active ``Context`` for this traced execution. This method is
-        automatically called in the ``tracer.trace()``, but it can be used in the application
-        code during manual instrumentation like::
+        # type (...) -> Context
+        """Return the active ``Context`` for the current execution."""
+        active = self.context_provider.active(*args, **kwargs)
+        if isinstance(active, Context):
+            return active
+        elif isinstance(active, Span):
+            return active.context
+        else:
+            return Context()
 
-            from ddtrace import tracer
+    def activate(self, span_or_ctx):
+        # type: (Union[Context, Span]) -> None
+        """Activate a span or context for the current execution context."""
+        self.context_provider.activate(span_or_ctx)
 
-            async def web_handler(request):
-                context = tracer.get_call_context()
-                # use the context if needed
-                # ...
+    def active(self):
```

Comment:
```suggestion
    def _active(self):
```

the public api that users would need is current_span